### PR TITLE
sast-unicode-check: add support for Trusted Artifacts

### PR DESCRIPTION
**Pull request by kdudka**
Create the `-oci-ta` variant of the `sast-unicode-check` task.

Related: https://github.com/konflux-ci/build-definitions/pull/1401
Resolves: https://issues.redhat.com/browse/OSH-764

### CODE REVIEW

**Comment by kdudka** [comment]
@rhyw Added.  Thanks for the notice!

**Comment by kdudka** [comment]
I have tested this via https://github.com/kdudka/konflux-test-ec-cli/pull/1/checks?check_run_id=31993329561.

**Comment by kdudka** [comment]
@hongweiliu17 @tkdchen Could you please move this forward?

The first commit of this pull request fixes #1401, which was merged broken.

**Comment by kdudka** [comment]
@hongweiliu17 There was a successful CI run yesterday: https://github.com/konflux-ci/build-definitions/runs/33305152289

How come it is now red again?

**Review by tkdchen — COMMENTED** [review]
LGTM generally. It's hard for me to review the logic of the script implementation since I'm not in the specific field.

**Comment by kdudka** [comment]
`sast-unicode-check-oci-ta` is generated from `sast-unicode-check` using the [hack/generate-ta-tasks.sh](https://github.com/konflux-ci/build-definitions/blob/bf3a7f2c00c3863ca68566942d1a40c43175bcb5/hack/generate-ta-tasks.sh) script.

File: task/sast-unicode-check-oci-ta/0.1/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+Scans source code for non-printable unicode characters in all text files.
+
+## Parameters
+|name|description|default value|required|
+|---|---|---|---|
+|CACHI2_ARTIFACT|The Trusted Artifact URI pointing to the artifact with the prefetched dependencies.|""|false|
+|FIND_UNICODE_CONTROL_ARGS|arguments for find-unicode-control command.|-p bidi -v -d -t|false|
+|FIND_UNICODE_CONTROL_GIT_URL|URL from repository to find unicode control.|https://github.com/siddhesh/find-unicode-control.git#c2accbfbba7553a8bc1ebd97089ae08ad8347e58|false|
+|KFP_GIT_URL|URL from repository to download known false positives files.|""|false|
+|PROJECT_NVR|Name-Version-Release (NVR) of the scanned project. It is used to find path exclusions (it is optional). |""|false|
```
Comment (14rcole) [inline task/sast-unicode-check-oci-ta/0.1/README.md:12]: In discussions about other sast checks we agreed to use `PROJECT_NAME` and have it default to the component name.  At build-time, components won't have an NVR.

Ref: https://github.com/konflux-ci/build-definitions/pull/1437/files#diff-4b781f85768af90bc67dfb600a14fb9e1819077a3c0acca7ef553aa9b6107478R12

File: task/sast-unicode-check-oci-ta/0.1/sast-unicode-check-oci-ta.yaml [hunk 2]
```diff
@@ -0,0 +1,287 @@
...
+        GIT_URL=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $1}')
+        REV=$(echo "${FIND_UNICODE_CONTROL_GIT_URL}" | awk -F'#' '{print $2}')
+
+        # Clone find-unicode-control repository
+        if ! git clone "${GIT_URL}" find-unicode-control; then
+          echo "Failed to clone the repository: ${GIT_URL}" >&2
+          note="Task $(context.task.name) failed: For details, check Tekton task log."
+          ERROR_OUTPUT=$(make_result_json -r ERROR -t "$note")
+          echo "${ERROR_OUTPUT}" | tee "$(results.TEST_OUTPUT.path)"
+          exit 1
```
Comment (hongweiliu17) [inline task/sast-unicode-check-oci-ta/0.1/sast-unicode-check-oci-ta.yaml:141]: All `exit 1` should be changed to `exit 0` then pipeline can proceed according to https://github.com/konflux-ci/architecture/blob/main/ADR/0014-let-pipelines-proceed.md#decision, and EC will check the result in `TEST_OUTPUT`.

**Comment by hongweiliu17** [comment]
Comment for `exit 1` according to https://github.com/konflux-ci/architecture/blob/main/ADR/0014-let-pipelines-proceed.md#decision. Others look good to me.
It is better to test the task before merging in case any issues.

File: task/sast-unicode-check-oci-ta/0.1/sast-unicode-check-oci-ta.yaml (hunk 2 above)
Comment (dirgim) [reply task/sast-unicode-check-oci-ta/0.1/sast-unicode-check-oci-ta.yaml:141]: Shouldn't the trap directive [here](https://github.com/konflux-ci/build-definitions/pull/1524/files#diff-6916740abc2ca63e64ccb9fc5bfed61b7bb78cdb8ab747cad37a49d7dcc78ad5R114) catch the exit 1 calls?

Comment (hongweiliu17) [reply task/sast-unicode-check-oci-ta/0.1/sast-unicode-check-oci-ta.yaml:141]: Ah, yes, you are right!

**Review by hongweiliu17 — APPROVED** [review]
lgtm

**Review by dirgim — APPROVED** [review]
LGTM code-wise

**Comment by kdudka** [comment]
I can see this is approved and CI checks are all green.  @14rcole Could we please proceed with merging this?

**Pull request merged**